sql/src/main/java/io/crate/expression/reference/sys/job/JobContext.java

```diff
@@ -22,21 +22,27 @@
 package io.crate.expression.reference.sys.job;
 
 import io.crate.auth.user.User;
+import io.crate.planner.operators.StatementClassifier.Classification;
 
+import javax.annotation.Nullable;
 import java.util.UUID;
 
 public class JobContext {
 
-    public final UUID id;
-    public final String username;
-    public final String stmt;
-    public final long started;
+    private final UUID id;
+    private final String username;
+    private final String stmt;
+    private final long started;
+    @Nullable
+    private final Classification classification;
 
-    public JobContext(UUID id, String stmt, long started, User user) {
+    public JobContext(UUID id, String stmt, long started, User user,
+                      @Nullable Classification classification) {
```

Comment:
In which case can this be nullable?

https://github.com/crate/crate/pull/7366/files#diff-a5be4157de855087dc9e01bcd7b28686R105 doesn't seem to allow nulls and the classifier always returns some classification?

Reply:
In case of an error during planning resulting in `logPreExecutionFailure()` https://github.com/crate/crate/pull/7366/files#diff-a5be4157de855087dc9e01bcd7b28686R150
I was thinking whether to use the `UNDEFINED` type for the classification then, but because it won't be further used, I decided to make it `null`.